src/hooks/useFiles.ts

```diff
@@ -16,45 +15,34 @@ export function useAddFiles (dispatch: Dispatch<FilesAction>, heliaState: HeliaC
   if (heliaState.starting) {
     throw new Error('Helia not active')
   }
-  const { helia, mfs } = heliaState
+  const { mfs } = heliaState
 
   return (files: File[]) => {
     for (const _file of files) {
-      const id: string = shortid.generate()
+      // const id: string = shortid.generate()
       const name = _file.name
 
-      const file: AddFileState = {
-        id,
-        name,
-        size: _file.size,
-        progress: 0,
-        cid: null,
-        pending: true,
-        published: false
-      }
-
       Promise.resolve().then(async () => {
-        dispatch({ type: 'add_start', ...file })
         const content = blobToIt(_file)
         await mfs.writeByteStream(content, name)
         const { cid } = await mfs.stat(`/${name}`)
-        dispatch({ type: 'add_success', id, cid })
+
+        const file: AddFileState = {
+          id: cid.toString(),
+          name,
+          size: _file.size,
+          progress: 0,
+          cid,
+          pending: true,
+          published: false
+        }
+        dispatch({ type: 'add_start', ...file })
```

Comment:
We did the add_start event move down?

Reply:
I moved it down because I updated the fileProvider to use the same publishing logic, and use CIDs for the files property on the fileProvider state. So instead of `Record<string, FileState>` with a `shortid` it's now `Record<CID.toString(), FileState>`, and add_start was triggering publish logic before we had the CID (with `add_success`). 

Since this is all in the browser now (not using RPC calls to a kubo node) we don't need to do `add_start` and `add_success`.. I need to change these to upload_start and upload_success or similar, and probably add another property on the filesProvider for `filesToUpload` to separate them from the `files` Record